crates/worker/src/job_manager.rs

```diff
@@ -110,20 +135,38 @@ impl JobManager {
                     self.network.clone(),
                     self.work_dir_base.clone(),
                     config,
-                    self.config.clone(),
                 );
                 let execution = executor
                     .execute(spec.clone(), cancel_token.clone(), spec.job_id, scheduler)
                     .await?;
+
+                let jobs = self.jobs.clone();
+                let monitor = tokio::spawn(async move {
+                    let status = match execution.wait().await {
+                        Ok(s) => s,
+                        Err(e) => Status::Failed(e.to_string()),
+                    };
+
+                    tracing::info!("Job completed with status: {:?}", status);
```

Comment:
Nit: Should we include the job ID here?

```suggestion
                    tracing::info!(job_id = spec.job_id, "Job completed with status: {:?}", status);
```

Reply:
Yes, good idea!

Reply:
Added

Reply:
Looks the changes haven't been added/pushed?